src/updater.ts

```diff
@@ -0,0 +1,114 @@
+import { Telegraf, Context } from 'telegraf'
+import { ToadScheduler, SimpleIntervalJob, AsyncTask } from 'toad-scheduler'
+import axios from 'axios'
+import execa from 'execa'
+import { getAdmin } from './bot/isAdmin'
+
+export async function setupAutoUpdate(bot: Telegraf<Context>) {
+    const scheduler = new ToadScheduler()
+    const task = new AsyncTask(
+        'check for new release',
+        () => {
+            return checkForUpdate(bot)
+        },
+        (err: Error) => {
+            bot.telegram.sendMessage(
+                getAdmin(),
+                `Error in update ${err.name}\n${err?.message}`
+            )
+        }
+    )
+
+    if (process.env.UPDATE_INTERVAL !== '0') {
+        const intervalString = process.env.UPDATE_INTERVAL || '1d'
+
+        const interval = convertToScheduleerInterval(intervalString)
+
+        const job = new SimpleIntervalJob(
+            { ...interval, runImmediately: true },
+            task
+        )
+
+        scheduler.addSimpleIntervalJob(job)
+    }
+}
+
+function convertToScheduleerInterval(intervalString: string) {
+    const lastLetter = intervalString[intervalString.length - 1]
+    const timeFrame = convertLetterToTimeFrame(lastLetter)
+    const intervalAmount = Number(
+        intervalString.substring(0, intervalString.length - 2)
+    )
+    if (isNaN(intervalAmount)) {
+        throw new Error(
+            `Unkown inteval "${intervalAmount}" please use a known pattern like "1d"`
+        )
+    }
+    return { [timeFrame]: intervalAmount }
+}
+
+function convertLetterToTimeFrame(
+    letter: string
+): 'seconds' | 'minutes' | 'hours' | 'days' {
+    switch (letter) {
+        case 's':
+            return 'seconds'
+        case 'm':
+            return 'minutes'
+        case 'h':
+            return 'hours'
+        case 'd':
+            return 'days'
+        default:
+            console.warn(
+                `Uknown interval letter "${letter}" please use d\\h\\m\\s`
+            )
+            return 'days'
+    }
+}
+
+type Releases = Release[]
+interface Release {
+    body: string | undefined
+    published_at: string | undefined
+    created_at: string | undefined
+    tag_name: string | undefined
+    name: string | undefined
+}
+
+async function checkForUpdate(bot: Telegraf<Context>) {
+    // TODO: Maybe get the url from origin remote and fallback to deanshub/downloader?
```

Comment:
@Schniz are you using a fork? do you need this?

Reply:
You can add an env var for overriding it. No need for hard git stuff. I use upstream